src/socks-common/socks-headers.ts

```diff
@@ -176,7 +176,11 @@ export function composeAuthHandshakeBuffer(auths:Auth[]) : ArrayBuffer {
   var handshakeBytes = new Uint8Array(auths.length + 2);
   handshakeBytes[0] = VERSION5;
   handshakeBytes[1] = auths.length;
-  handshakeBytes.set(auths, 2);
+  // TODO: Revert to set() when Uint8Array's type definition is fixed in TS1.6:
```

Comment:
Right. I couldn't find a way to augment the `Uint8Array` definition so I took your first suggestion.